File: Brackets.py
string = "(([{}](){}))"


class Stack:
    def __init__(self):
        self.items = []

    def isEmpty(self):
        return self.items == []

    def push(self, item):
        self.items.append(item)

    def pop(self):
        return self.items.pop()

    def peek(self):
        return self.items[len(self.items)-1]

    def size(self):
        return len(self.items)


def closed(text):
    if len(text) % 2 != 0:
        return False
    k = text[0]
    if k == ']' or k == '}' or k == ')':
        return False
    #print(k)

    s = Stack()
    for char in text:
        if char == '[' or char == '{' or char == '(':
            s.push(char)

        if char == ']' or char == '}' or char == ')':
            s.pop()
    return True

print(closed(string))

File: Brackets.java
import java.util.Stack;

public class Brackets {

    public static void main(String[] args) {
        String s1 = "(([{}](){}))";
        System.out.println(isClosed(s1));

    }

    public static boolean isClosed(String text) {
        //Base cases
        if(text.length() % 2 != 0) return false;
        char k = text.charAt(0);
        if (k == ']' || k == '}' || k == ')') return false;

        Stack<Character> stack = new Stack<Character>();
        for(int i = 0; i < text.length(); i++) {

            char current = text.charAt(i);
            if(current == '[' || current == '{' || current == '(') //If opening brackets,
                stack.push(current);                                //then push onto the stack


            if (current == ']' || current == '}' || current == ')')
                stack.pop();
        }
        return true;
    }

}
